### PR TITLE
Throttle scroll event 

### PR DESCRIPTION
**Pull request by shavit**
This decreases the amount of function calls when scrolling through the documentation website.

# Description

Two scroll events continuously check the DOM for changes and highlight navigation links. The change decreases the number of function calls.

It is possible to improve usability and SEO by creating static documentation pages.  Each section can have a dedicated page with its unique title, and this will also help with runtime issues on the page.

Related:
- https://github.com/gradio-app/gradio/issues/3181
- https://github.com/gradio-app/gradio/issues/3267 ("Smooth scrolling", "Scrolling highlight in nav in guides, docs and changelog")
- https://github.com/gradio-app/gradio/issues/3840

# Checklist:

- [ x] I have performed a self-review of my own code
- [ x] I have added a short summary of my change to the CHANGELOG.md
- [ x] My code follows the style guidelines of this project
- [ x] I have commented my code in hard-to-understand areas
- ~[ ] I have made corresponding changes to the documentation~
- ~[ ] I have added tests that prove my fix is effective or that my feature works~
- [ x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -48,6 +48,7 @@ No changes to highlight.
 - Fix `gr.Slider` `release` event not triggering on mobile by [@space-nuko](https://github.com/space-nuko) in [PR 4098](https://github.com/gradio-app/gradio/pull/4098) 
 - Removes extraneous `State` component info from the `/info` route by [@abidlabs](https://github.com/freddyaboulton) in [PR 4107](https://github.com/gradio-app/gradio/pull/4107)
 - Make .then() work even if first event fails by [@aliabid94](https://github.com/aliabid94) in [PR 4115](https://github.com/gradio-app/gradio/pull/4115).
+- Scroll event performance by [@shavit](https://github.com/shavit) in [PR 4158](https://github.com/gradio-app/gradio/pull/4158).
```
Comment (pngwn) [inline CHANGELOG.md:51]: This change does not require a changelog entry.

Comment (shavit) [reply CHANGELOG.md]: Alright.

**Comment by pngwn** [comment]
Will take a look at this soon.

**Review by aliabd — APPROVED** [review]
Everything looks good! nice work @shavit. Made a minor fix unrelated to you.

**Pull request merged**